### DISCUSSION TITLE
How would I create a terminal with a vertical split?

### DISCUSSION THREAD

**Discussion by jjohnston-aims** [folke/snacks.nvim#68] [Q&A] [answered]
Maybe this should be in the Edgy repo, but how do I create a terminal with a vertical split?

**Comment by iamxiaojianzheng** [comment]
Try this

```lua
{
    "folke/snacks.nvim",
    opts = {
      terminal = {
        win = {
          position = "right",
        },
      },
    }
}
```

**Reply by jjohnston-aims** [reply]
That works well as a permanent change. How would I do it for one terminal? I thought I could do something like this, but it doesn't seem to work:

`:lua Snacks.terminal.open(opts = { win = { position = "right" }})`

**Reply by jjohnston-aims** [reply]
Also, the config overwrites the terminal position when cmd is provided, which makes lazygit open on the right.

**Reply by iamxiaojianzheng** [reply]
> Also, the config overwrites the terminal position when cmd is provided, which makes lazygit open on the right.

I talked to folke about this, and we'll see what he says.

**Reply by iamxiaojianzheng** [reply, marked as answer]
> That works well as a permanent change. How would I do it for one terminal? I thought I could do something like this, but it doesn't seem to work:
> 
> `:lua Snacks.terminal.open(opts = { win = { position = "right" }})`


This work for me

```lua
lua Snacks.terminal.open({"bash"}, { win = { position = "right" }})
```

**Reply by rafgugi** [reply]
> This work for me
> 
> ```lua
> lua Snacks.terminal.open({"bash"}, { win = { position = "right" }})
> ```

or don't pass the cmd
```lua
lua Snacks.terminal.open(nil, { win = { position = "right" }})
```

**Reply by LintaoAmons** [reply]
Is there a way to find and open the hidden terminals?
